spec.md

```diff
@@ -1292,8 +1293,8 @@ The CO MUST implement the specified error recovery behavior when it encounters t
 
 | Condition | gRPC Code | Description | Recovery Behavior |
 |-----------|-----------|-------------|-------------------|
-| Volume does not exist | 5 NOT_FOUND | Indicates that a volume corresponding to the specified `volume_id` does not exist. | Caller MUST verify that the `volume_id` is correct and that the volume is accessible and has not been deleted before retrying with exponential back off. |
-| Node does not exist | 5 NOT_FOUND | Indicates that a node corresponding to the specified `node_id` does not exist. | Caller MUST verify that the `node_id` is correct and that the node is available and has not been terminated or deleted before retrying with exponential backoff. |
+| Volume does not exist and volume not assumed ControllerUnpublished from node | 5 NOT_FOUND | Indicates that a volume corresponding to the specified `volume_id` does not exist and volume not assumed ControllerUnpublished from node. | Caller MUST verify that the `volume_id` is correct and that the volume is accessible and has not been deleted before retrying with exponential back off. |
```

Comment:
Could we get a little more verbose in the "Description" column? Specifically, how about this?

> Indicates that a volume corresponding to the specified `volume_id` does not exist and is not assumed to be ControllerUnpublished from the node. 

or with some emphasis,

> Indicates that a volume corresponding to the specified `volume_id` does not exist **and is not assumed to be ControllerUnpublished from the node**.

Reply:
I think the current external_attacher is also not implementing Recovery Behavior which a "MUST" in the spec.

Reply:
external-attacher I think is implementing the recovery behavior to the best of its ability based off of the state that it stores.  Are you suggesting to change the recovery behavior wording? cc @jsafrane

Reply:
The external-attacher checks that the node exists (i.e. Kubernetes CSINode or Node objects exist).

The external-attacher checks that Kubernetes object for the volume (PersistentVolume) exists. It does **not** verify that the volume exists in SP. The only way is `ListVolumes`, which I find quite clumsy, there is no filtering there.

However, even if the attacher checked that node / volume does did not exist, the spec not say if it can assume the volume has been ControllerUnpublished and what are required cleanup actions, so this check would be quite useless.

Reply:
Improved the wording as per @misterikkit suggestion. Agree with @msau42 and @jsafrane that we already implement the spirit of the behavior through stored data, as well as not doing part of it explicitly for good reason.